mne/preprocessing/maxwell.py

```diff
@@ -146,39 +152,30 @@ def maxwell_filter(raw, origin=(0., 0., 40.), int_order=8, ext_order=3,
 
     # Fine calibration processing (point-like magnetometers and calib. coeffs)
     if fine_cal is not None:
-        logger.info('Fine calibration file used in Maxwell filter: ' +
-                    fine_cal)
-
         mag_inds = pick_types(raw_sss.info, meg='mag')
         grad_inds = pick_types(raw_sss.info, meg='grad')
         grad_info = pick_info(raw_sss.info, grad_inds)
 
         # Compute point-like mags to incorporate gradiometer imbalance
-        grad_imbalances = np.array([ch['calib_coeff'] for ch in cal_chans
-                                    if ch['ch_type'] == 'grad'])
-        S_in_fine, S_out_fine = _sss_basis_point(origin, grad_info, int_order,
-                                                 ext_order, grad_imbalances)
+        grad_imbalances = np.array([cal_chans[ii]['calib_coeff']
```

Comment:
`cal_chans` is guaranteed now to be in the same order as the channels in the acquisiton file (something that was assumed but not guaranteed before), so we can just use the same `picks` to get the right vals